### ISSUE TITLE
Pages with 2+ password fields must use different programmatic labels for 'show' and 'hide' buttons on those password fields

### ISSUE THREAD

**Issue by davidc-gds** [alphagov/govuk-frontend#3675]
This issue is from May 2023 external accessibility audit report.

**WCAG Reference:**
2.4.6 Headings and Labels (Level AA)
[Understanding Headings and Labels](https://www.w3.org/WAI/WCAG21/Understanding/headings-and-labels.html) | [How to Meet Headings and Labels](https://www.w3.org/WAI/WCAG21/quickref/#headings-and-labels)

**Issue ID:**
DAC_Non-descriptive_Labels_01

**URLs:**
URL: https://signin.account.gov.uk/create-password

## Screen shots
<img width="882" alt="image" src="https://github.com/alphagov/govuk-frontend/assets/98318778/f2138cf5-cb3d-40af-b509-a00911d69db6">

Both of the ‘Show’ buttons presented for users have been provided with an ‘aria-label’ to offer users of screen reading assistive technologies additional context that they are to show the password. However, both buttons have been provided the same name of ‘Show password’, meaning that screen reader users are not able to distinguish which button relates to which field.
This same occurs when the button is activated, and the label is changed to ‘hide’.

## Current code ref(s)
`#main-content > div > div > form > div:nth-child(3) > div > div > button`

```<button class="govuk-show-password__toggle" aria-controls="password" type="button" aria-label="Show password">Show</button> […] <button class="govuk-show-password__toggle" aria-controls="confirm-password" type="button" aria-label="Show password">Show</button>```

## Screen-reader comments
> “I found that two buttons announcing as ‘Show password’ were present on the page which when selected updated to read as ‘Hide password’. The label of each button was duplicated but related to different password fields. It was not immediately clear when situated out of context which button related to which password field potentially leading to the user making an incorrect selection. Although I was not personally affected in this instance some screen reader users may become confused. Giving each button label unique content will enable all users to make an informed selection both in and out of context.”

## Solution
Ensure that labels are descriptive enough for users to identify their purpose and to which content they relate. The label for each ‘show’ button should inform users of screen reading assistive technologies to which field they relate.
The second button should have an ‘aria-label’ value of ‘Show re-type password’ for example.
